### PR TITLE
Logan dev

### PR DESCRIPTION
**Pull request by logyball**
added an initial implementation of the board as well as the "move".  assuming most of the is it my turn/can i move this slot is checked at the client for now.  Added a simple travis yaml

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 > Client/Server application to play Mancala written in Rust.
 
+![rusty mancala](./img/rust-mancala.jpg "")
```
Comment (rooneyshuman) [inline README.md:5]: this is important. thanks for adding

File: mancala/protocol/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    player_one_goal_slot: usize,
+    player_two_goal_slot: usize,
+    player_one_turn: bool,
+}
+
+impl GameState {
+    fn new(p_one: String, p_two: String) -> GameState {
+        let mut init_game_board = [STARTING_STONES; SLOTS * 2];
+        init_game_board[SLOTS + 1] = 0;
+        init_game_board[0] = 0;
```
Comment (rooneyshuman) [inline mancala/protocol/src/lib.rs]: are these initializing the goal slots? if so, line 45 should be `init_game_board[SLOTS] = 0;`

File: mancala/protocol/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            let slot_to_add_to: usize = cur_slot % board_length;
+            if slot_to_add_to == goal_slot {
+                self.add_points(1);
+                continue;
+            }
+            self.game_board[slot_to_add_to] += 1;
+        }
+        self.player_one_turn = !self.player_one_turn;
+    }
+}
```
Comment (rooneyshuman) [inline mancala/protocol/src/lib.rs]: need to add a check if `cur_slot` is the opposing player's goal_slot. If so, skip that slot and don't add a stone to it.

**Review by rooneyshuman — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (logyball) [reply README.md:5]: 👍

File: mancala/protocol/src/lib.rs (hunk 2 above)
Comment (logyball) [reply mancala/protocol/src/lib.rs]: yup good catch

File: mancala/protocol/src/lib.rs (hunk 3 above)
Comment (logyball) [reply mancala/protocol/src/lib.rs]: another good catch, ill put in a fix to make this make more sense

**Comment by logyball** [comment]
updated based on comments - nice catches.  Made the struct public as well because it should be public, simplified the logic for making a move greatly

**Review by rooneyshuman — APPROVED** [review]
Thanks for updating the tests too! Looks good
Alt Text

**Pull request merged**